tee-worker/litentry/core/credentials/src/credentials.rs

```diff
@@ -0,0 +1,269 @@
+// Copyright 2020-2022 Litentry Technologies GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+use crate::error::Error;
+use chrono::{DateTime, FixedOffset, NaiveDateTime, SecondsFormat};
+use codec::{Decode, Encode};
+use itp_types::AccountId;
+use itp_utils::stringify::account_id_to_string;
+use litentry_primitives::Assertion;
+use log::*;
+use scale_info::TypeInfo;
+use serde::{Deserialize, Serialize};
+use std::{
+	fmt::Debug,
+	string::{String, ToString},
+	time::{SystemTime, UNIX_EPOCH},
+	vec::Vec,
+};
+
+pub const PROOF_PURPOSE: &str = "assertionMethod";
+pub const MAX_CREDENTIAL_SIZE: usize = 2048;
+
+pub fn now() -> String {
+	let ts = SystemTime::now().duration_since(UNIX_EPOCH).unwrap();
+	let naive = NaiveDateTime::from_timestamp_opt(ts.as_secs() as i64, ts.subsec_nanos()).unwrap();
+	let datenow_time = DateTime::<FixedOffset>::from_utc(naive, FixedOffset::east_opt(0).unwrap());
+	datenow_time.to_rfc3339_opts(SecondsFormat::Secs, true)
+}
+
+#[derive(Serialize, Deserialize, Encode, Decode, Clone, Debug, PartialEq, Eq, TypeInfo)]
+pub enum ProofType {
```

Comment:
We'll need quite some comments in this file, as it's the core part of the VC format.

E.g. explanations of `ProofType`, `CredentialType`: what do they exactly mean and what are the possible values? Same applies to other important structs/fields.

Reply:
added